dhis-2/dhis-services/dhis-service-core/src/test/java/org/hisp/dhis/trackedentity/TrackedEntityAttributeStoreIntegrationTest.java

```diff
@@ -0,0 +1,212 @@
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.trackedentity;
+
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.hisp.dhis.IntegrationTestBase;
+import org.hisp.dhis.common.ValueType;
+import org.hisp.dhis.program.Program;
+import org.hisp.dhis.program.ProgramService;
+import org.hisp.dhis.program.ProgramTrackedEntityAttribute;
+import org.hisp.dhis.security.acl.AccessStringHelper;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * @author Chau Thu Tran
+ */
+public class TrackedEntityAttributeStoreIntegrationTest
+    extends
+    IntegrationTestBase
+{
+    @Autowired
+    private TrackedEntityAttributeService attributeService;
+
+    private TrackedEntityAttribute attributeW;
+
+    private TrackedEntityAttribute attributeY;
+
+    private TrackedEntityAttribute attributeZ;
+
+    @Autowired
+    private TrackedEntityTypeService trackedEntityTypeService;
+
+    @Autowired
+    private ProgramService programService;
+
+    private final static int A = 65;
+
+    private final static int T = 85;
+
+    private Program programB;
+
+    @Override
+    public void setUpTest()
+    {
+
+        attributeW = createTrackedEntityAttribute( 'W' );
+        attributeW.setUnique( true );
+        attributeY = createTrackedEntityAttribute( 'Y' );
+        attributeY.setUnique( true );
+        attributeZ = createTrackedEntityAttribute( 'Z', ValueType.NUMBER );
+
+        List<TrackedEntityAttribute> attributesA = new ArrayList<>();
+        attributesA.add( attributeW );
+        attributesA.add( attributeY );
```

Comment:
is this ever used? I assume this is what the code scanning says 🤔

Reply:
Yes I am not done with changes. Clearing Sonar and CodeQL issues right now. You are just too quick to re-review 🥇

Reply:
haha, sorry. I will keep quiet know 😅  let me know when you need/want help and I'll support 😄